.github/workflows/release-azure-containerapp-v1.yaml

```diff
@@ -0,0 +1,376 @@
+
+          if [[ "$CURRENT_REVISION" == "$new_revision" ]]; then
+            echo "::warning::No deployment needed - current revision is already using the latest image"
+            echo "skip_remaining_steps=true" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          echo "new_revision_name=$new_revision" >> $GITHUB_OUTPUT
+          echo "skip_remaining_steps=false" >> $GITHUB_OUTPUT
+
+          if [[ "$REVISION_MODE" == "Multiple" ]]; then
+            az containerapp ingress traffic set \
+              --name "$CONTAINER_APP_NAME" \
+              --revision-weight "$CURRENT_REVISION"=100 \
+              --revision-weight $new_revision=0 \
+              --output table
+          fi
+
+      - name: Wait for the New Revision to be Ready
+        if: ${{ steps.create_new_revision.outputs.skip_remaining_steps == 'false' }}
+        env:
+          NEW_REVISION_NAME: ${{ steps.create_new_revision.outputs.new_revision_name }}
+        run: |
+          max_attempts=30
+          attempt=0
+          health_state=""
+
+          echo "::notice::Waiting for new revision ("$NEW_REVISION_NAME") to be healthy..."
+
+          while [ "$health_state" != "Healthy" ] && [ $attempt -lt $max_attempts ]; do
+
+            health_state=$(az containerapp revision list \
+              --name "$CONTAINER_APP_NAME" \
+              --query "[?contains(name, '$NEW_REVISION_NAME')].{Health:properties.healthState}|[0].Health" -o tsv)
+
+            echo "::notice::Current health state: $health_state (attempt $((attempt+1))/$max_attempts)"
+
+            if [ "$health_state" == "Healthy" ]; then
+              echo "::notice::New revision is now healthy!"
+              break
+            fi
+
+            attempt=$((attempt+1))
+            sleep 3
+          done
+
+          if [ "$health_state" != "Healthy" ]; then
+            echo "::error::New revision did not become healthy within the timeout period"
+            exit 1
+          fi
+
+      - name: Get Container App Id
+        id: get_container_app_id
+        if: ${{ steps.get_revision_mode.outputs.revision_mode == 'Multiple' && steps.create_new_revision.outputs.skip_remaining_steps == 'false' }}
+        run: |
+          container_app_id=$(az containerapp show \
+            --name "$CONTAINER_APP_NAME" \
+            --query "id" \
+            --output tsv)
+
+          echo "ca_id=$container_app_id" >> $GITHUB_OUTPUT
+
+      - name: Check for New Revision Errors and Increment Traffic
+        if: ${{ steps.get_revision_mode.outputs.revision_mode == 'Multiple' && steps.create_new_revision.outputs.skip_remaining_steps == 'false' }}
+        env:
+          SCRIPT_NAME: ${{ inputs.script_name }}
+          NEW_REVISION_NAME: ${{ steps.create_new_revision.outputs.new_revision_name }}
+          CURRENT_REVISION_NAME: ${{ steps.get_current_revision.outputs.current_revision_name }}
+          FUNCTION_APP_NAME: "io-p-itn-auth-lollipop-func-02"
```

Comment:
suggestion: mark all the debug code in order to let reviewers tell it from the real one